pkg/checks/checks.go

```diff
@@ -30,23 +25,8 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}
 	return wait.PollUntil(time.Second, func() (bool, error) {
 		resp, err := cli.Do(req)
-		if err, ok := err.(*url.Error); ok {
-			if err, ok := err.Err.(*net.OpError); ok {
-				if err, ok := err.Err.(*os.SyscallError); ok {
-					if err.Err == syscall.ENETUNREACH {
-						return false, nil
-					}
-				}
-			}
-			if err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF {
-				return false, nil
-			}
-		}
-		if err == io.EOF {
-			return false, nil
-		}
 		if err != nil {
-			return false, err
+			return false, nil
 		}
 		return resp != nil && resp.StatusCode == http.StatusOK, nil
 	}, ctx.Done())
```

Comment:
The Background context is never cancelled so essentially this will loop forever until the apiserver is up. We may want to do something different in a follow-up.